src/components/queryExpressionEditor/expressionEditor/ExpressionEditorConnected.tsx

```diff
@@ -0,0 +1,45 @@
+import {connect} from 'react-redux';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../../ReactVapor';
+import {IReduxAction, ReduxUtils} from '../../../utils/ReduxUtils';
+import {ICheckboxActionPayload} from '../../checkbox/CheckboxActions';
+import {IDropdownOption} from '../../dropdownSearch/DropdownSearch';
+import {IDropdownSearchState} from '../../dropdownSearch/DropdownSearchReducers';
+import {IListBoxState} from '../../listBox/ListBoxReducers';
+import {FIELD_SELECT_ID} from '../fieldSelect/FieldSelect';
+import {OPERATOR_SELECT_ID} from '../operatorSelect/OperatorSelect';
+import {ExpressionEditor, IExpressionEditorDispatchProps, IExpressionEditorOwnProps, IExpressionEditorProps, IExpressionEditorStateProps} from './ExpressionEditor';
+import {removeExpressionEditor, updateExpressionEditor} from './ExpressionEditorActions';
+import {IExpressionEditorState} from './ExpressionEditorReducers';
+
+// TODO QUESTION R-V :
+// Is this the only way to get the selected option from a DropdownSearchConnected selector?
+// Note : Duplication of code with ValueSelectStringConnected.tsx
+function getSelectedOption(options: IDropdownOption[]): string {
+    const selectedOption: IDropdownOption = _.findWhere(options, {selected: true});
+    return selectedOption.value;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IExpressionEditorOwnProps): IExpressionEditorStateProps => {
+    const expressionEditors: IExpressionEditorState[] = state.expressionEditors && state.expressionEditors;
+    const expressionEditor: IExpressionEditorState = _.findWhere(state.expressionEditors, {id: ownProps.id});
+    const fieldSelect: IDropdownSearchState = _.findWhere(state.dropdownSearch, {id: `${ownProps.id}-${FIELD_SELECT_ID}`});

```

Comment:
you should have a method that retrieve child ids for fieldSelect and operatorSelect, we a caught a lot of bugs in the past because we were writing ids composition by hand like this. This is error prone. You can check `TableUtils` for examples.